control-plane/typeSpecConverter.cpp

```diff
@@ -111,6 +115,52 @@ bool TypeSpecConverter::preorder(const IR::Type_Name* type) {
     return false;
 }
 
+bool TypeSpecConverter::preorder(const IR::Type_Newtype* type) {
+    if (p4RtTypeInfo) {
+        bool orig = true;
+        cstring uri;
+        uint32_t sdnB;
+        auto ann = type->getAnnotation("p4runtime_translation");
```

Comment:
can't the `ParseAnnotations` pass help with validating / parsing the annotations?

Reply:
At least this code needs to parse an annotation to detect translated vs. original type.  How does the ParseAnnotations pass provide the information to the code here?  I am to understand such mechanics to see what we can do.

Reply:
@antoninbas Please see new checkin: https://github.com/p4lang/p4c/pull/1736/commits/379ca99aa5fdae7cd743323c61eeb5b971e56b5d

I added the p4runtime_translation to common Annotation validation infra.  Now, the p4c/control-plane/ code is spared from validity checks on this annotation and just use it.  This is the best I can do thus far.  I will beef up the validitiy check in the new .h file next.